### PR TITLE
Accept ' ' as a valid format specifier name

### LINKED ISSUE #2058: `' '` is not accepted as a valid format specifier key name
When type-checking the following code:
```python
print("%( )s world" % {' ': 'hello'})
```
`mypy` gives the error:
```
error: Unsupported format character '('
```

Sounds like mypy doesn't currently accept spaces as the keys for format specifiers.

### PR DESCRIPTION
**Pull request by ddfisher**
Fix #2058.

### CODE REVIEW

File: mypy/checkstrformat.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def check_str_interpolation(self, str: StrExpr, replacements: Node) -> Type:
         return self.named_type('builtins.str')
 
     def parse_conversion_specifiers(self, format: str) -> List[ConversionSpecifier]:
-        key_regex = r'(\((\w*)\))?'  # (optional) parenthesised sequence of characters
+        key_regex = r'(\(([\w ]*)\))?'  # (optional) parenthesised sequence of characters
```
Comment (gvanrossum) [inline mypy/checkstrformat.py]: Why not "anything except parentheses"? What does Python itself implement or document?

Comment (ddfisher) [inline mypy/checkstrformat.py]: The documentation doesn't specifically discuss this, but I tested all ASCII characters and everything except parentheses worked.  I'll fix this up.

Comment (gvanrossum) [inline mypy/checkstrformat.py]: OK, I already merged, so another PR will be needed.

Comment (ddfisher) [inline mypy/checkstrformat.py]: I decided to just change it. >_>
https://github.com/python/mypy/commit/7c766098824a7ea5ceaeb39cdcf5169d3cd75b9c

**Pull request merged**